docs/swagger-doc-coverage.md

```diff
@@ -0,0 +1,178 @@
+# Adding REST endpoints to Swagger UI
+
+[Swagger UI](http://swagger.io/swagger-ui/) is a graphical front-end that displays
+REST API documentation with working examples. It uses a description of the API
+defined according to the Swagger/OpenAPI specification. For Zulip, this is in
+`/static/yaml/zulip.yaml`, and the generated API reference is available at a URL
+similar to other Zulip documentation. You can find an example at
+[/apidocs](https://zulipchat.com/apidocs/). This page is a basic introduction to the
+format of `zulip.yaml` and how to add additional endpoints to the Swagger UI REST
+API reference.
+
+In a Swagger specification file, every configuration section is an object.
+Objects may contain other objects, or reference objects defined elsewhere. Larger
+API specifications may be split into multiple files. There are more types of
+objects than mentioned here, you can find the complete details at
+[Swagger/OpenAPI specification page](http://swagger.io/specification).
+
+## Working with the `zulip.yaml` file
+
+A Swagger specification file has three general parts: information and configuration,
+endpoint definitions, and object schemas referenced by other objects (as an alternative
+to defining everything inline.)
+
+### Configuration
+
+These objects, at the top of `zulip.yaml`, identify the API, define the backend host
+for the working examples, list supported schemes and types of authentication, and
+configure other settings. Once defined, information in this section rarely changes.
+
+For example, the `swagger` and `info` objects look like this:
+```
+# Basic Swagger UI info
+swagger: '2.0'
+info:
+  version: '1.0.0'
+  title: Zulip REST API
+  description: Powerful open source group chat
+  contact:
+    url: https://zulip.org/
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+```
+
+### Endpoint definitions
+
+The Paths Object contains [Path Item Objects](http://swagger.io/specification/#pathItemObject)
+for each endpoint path, which describe in detail the methods and parameters it
+accepts and the responses it returns.
+
+There is one Path Item Object for each supported method, containing a
+[Parameters Object](http://swagger.io/specification/#parameterObject) describing
+the required and optional inputs. A
+[Response Object](http://swagger.io/specification/#responseObject) similarly specifies
+the content of the response. They may reference schemas from a global Definitions
+Object (see [Schemas](#schemas), below.)
+
+Example:
+
+The `/messages` endpoint (shown here simplified) accepts a POST with four
+parameters (one optional), expects HTTP Basic authentication, and returns a JSON
+response of type `MessageResponse`. The parameter type `formData` means parameters
+are expected as POST data, as you might do with `curl -d`.
+
+```
+paths:
```

Comment:
Mind updating the example with the latest changes in the YAML file? There are a couple things that have changed, and it could be confusing for beginners :)